src/Compute/Compute/Strategies/ComputeRp/VirtualMachineStrategy.cs

```diff
@@ -127,7 +130,13 @@ static class VirtualMachineStrategy
                         Priority = priority,
                         EvictionPolicy = evictionPolicy,
                         BillingProfile = (maxPrice == null) ? null : new BillingProfile(maxPrice),
-                        SecurityProfile = (encryptionAtHostPresent == true) ? new SecurityProfile(encryptionAtHost: encryptionAtHostPresent) : null,
+                        SecurityProfile = (encryptionAtHostPresent == true || enableVtpm != null || enableSecureBoot != null || securityType != null)
+                    ? new SecurityProfile
+                    {
+                        EncryptionAtHost = encryptionAtHostPresent,
+                        UefiSettings = (enableVtpm != null || enableSecureBoot != null) ? new UefiSettings(enableSecureBoot, enableVtpm) : null,
```

Comment:
Make sure to at least manually test the different combinations of different parameters being used here to ensure this works.

Reply:
Tested manually as well as added test cases